.github/workflows/sonar.yml

```diff
@@ -4,29 +4,29 @@ on:
   workflow_run:
     workflows: [Build]
     types: [completed]
+  push:
+    branches:
+      - main
 
 jobs:
   Sonar:
     runs-on: ubuntu-latest
-    if: github.event.workflow_run.conclusion == 'success'
+    if: github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.event == 'pull_request'
```

Comment:
do you need to kick it off with an `if github.event.workflow_run &&` I'm guessing this is null in the push case?

Reply:
good point